### PR TITLE
Add popup before secret delete

### PR DESCRIPTION
**Pull request by JoaoVictor6**
## Add popup before secret delete
This is my solution for issue #175 :)

### Goals
- [ ] Create a dialog
- [ ] Isolate logic for delete action in a component
- [ ] Improve types on delete scope

### Demo
https://user-images.githubusercontent.com/68869379/210918265-cfc7078f-2013-4b94-b131-d78ec7231f19.mp4
> The bug on the version history box on sidebar has because when recording the server wasn't running :p

### CODE REVIEW

File: frontend/components/dashboard/SideBar.tsx [hunk 1]
```diff
@@ -28,6 +29,10 @@ interface OverrideProps {
   pos: number;
   comment: string;
 }
+export interface DeleteRowFunctionProps { 
+  ids: string[];
```
Comment (JoaoVictor6) [inline frontend/components/dashboard/SideBar.tsx:33]: I centralize this type. I noticed an inconsistency in the data property. In all cases, this is one item, why is it an array and not an object?

File: frontend/public/locales/en/dashboard.json [hunk 2]
```diff
@@ -25,6 +25,11 @@
     "comments": "Comments & Notes",
     "personal-explanation": "This secret is personal. It is not shared with any of your teammates.",
     "generate-random-hex": "Generate Random Hex",
-    "digits": "digits"
+    "digits": "digits",
+    "delete-key-dialog": {
```
Comment (JoaoVictor6) [inline frontend/public/locales/en/dashboard.json:29]: this structure is valid?

Comment (mv-turtle) [reply frontend/public/locales/en/dashboard.json:29]: yeah, I think that's good. We will need to modify this to make it more consistent

File: frontend/components/dashboard/SideBar.tsx (hunk 1 above)
Comment (mv-turtle) [reply frontend/components/dashboard/SideBar.tsx:33]: This is very strange. Logically, it should be an ObjectId, I don't know why it is an array

**Review by mv-turtle — APPROVED** [review]
Looks really great! Thank you so much for the contribution :)

File: frontend/components/dashboard/SideBar.tsx (hunk 1 above)
Comment (JoaoVictor6) [reply frontend/components/dashboard/SideBar.tsx:33]: I made some refactoring in the sidebar component. maybe I can create a pr with this change for your review?

Comment (mv-turtle) [reply frontend/components/dashboard/SideBar.tsx:33]: @JoaoVictor6 Sounds good! Looking forward

**Pull request merged**